### DISCUSSION TITLE
How to compile "Ender3V2-MM-LA" from source (n00b)?

### DISCUSSION THREAD

**Discussion by mrelwood** [mriscoc/Ender3V2S1#492] [Q&A] [answered]
I would really love to try the Linear Advance feature since it sounds like it would be just the ticket to fix my print issues, but the available binaries are only for auto bed leveling sensors, which I don't have. My board is the non-silent 4.2.2., so unless I'm mistaken the LA should work with my printer. Tried the "Ender3V2-422-BLTUBL-LA-20221002.bin" which wasn't able to home the Z axis at all. So I would essentially need a "Ender3V2-MM-LA-2022xxxx.bin".

I am advanced with Macs, but I'm a total n00b when it comes to compiling from source. I may have ever succeeded only once with step by step instructions, but such instructions are not available for the MriscoC firmware. The existing discussions about compiling from source either didn't apply or went way over my head.

I do have Python3 installed, and I was able to run the following in Terminal:
```
#!/usr/bin/python
import CreateConfigs
CreateConfigs.Generate('Ender3V2-422-MM', ['Ender3V2','422','MM','LinAdv'])
```
which resulted in:

```
Configurations generator script for the Professional Firmware
Author: Miguel A. Risco Castillo (c) 2022

-Process Ender3V2-422-MM/Configuration.h
-Process Ender3V2-422-MM/Configuration_adv.h
-Process Ender3V2-422-MM/Version.h
-Process Ender3V2-422-MM/platformio.ini

Configuration files correctly generated
False
```

I don't know if "False" refers to the Configuration files being correctly generated, or to something else completely. Either way, it looks like the process didn't complete. It only created a log.txt in the Ender3V2-422-MM folder, which only says the same that's in the above quote, except without the "False".

I have absolutely no idea what I should or could try next, since the whole compiling process is alien to me. Help?

**Comment by mriscoc** [comment, marked as answer]
The last "false" statement can be the result of the last line in the Python script:
```python
return error
```
So, there isn't an error.

The generated configuration files must be copied to the main project folder according to: https://github.com/mriscoc/Special_Configurations

This is the MM version of the LA special configuration:
[Ender3V2-422-MM-LA.zip](https://github.com/mriscoc/Ender3V2S1/files/9953638/Ender3V2-422-MM-LA.zip)
